client/state/form/actions.js

```diff
@@ -15,67 +16,47 @@ export const setFormProperty = ( field, value ) => {
 	};
 };
 
-export const goToStep = ( stepIndex ) => {
-	return {
-		type: GO_TO_STEP,
-		step: stepIndex,
-	};
-};
-
-export const nextStep = () => ( dispatch, getState, { callbackURL, nonce, submitMethod, formSchema, formLayout } ) => {
-	const submitForm = ( callback ) => {
-		const setIsSaving = ( value ) => {
-			dispatch( setFormProperty( 'isSaving', value ) );
-			if ( false === value ) {
-				callback();
-			}
-		};
-		const setSuccess = ( value ) => dispatch( setFormProperty( 'success', value ) );
-		const setError = ( value ) => dispatch( setFormProperty( 'errors', value ) );
-
-		dispatch( setFormProperty( 'pristine', true ) );
-		saveForm( setIsSaving, setSuccess, setError, callbackURL, nonce, submitMethod, getState().form.values );
-	};
-
-	const tryAutoAdvance = ( isManualAction ) => {
-		const currentStep = getState().form.currentStep;
-		const currentStepLayout = formLayout[ currentStep ] || {};
+export const backFromSuggestion = () => ( {
+	type: BACK_FROM_SUGGESTION,
+} );
 
-		if ( currentStep >= formLayout.length - 1 ) {
-			return;
-		}
+export const goToStep = ( stepIndex ) => ( dispatch, getState, { formLayout } ) => {
+	const stepLayout = formLayout[ stepIndex ] || {};
 
-		if ( getStepFormErrors( getState(), formSchema, formLayout ).length ) {
-			return;
-		}
+	// Clear the "bypass suggestion" flag if the user is coming *back* to this step
+	if ( getState().form.currentStep > stepIndex && stepLayout.bypass_suggestion_flag ) {
+		dispatch( FormValueActions.updateField( stepLayout.bypass_suggestion_flag, false ) );
+	}
 
-		if ( getState().form.pristine ) {
-			dispatch( goToStep( currentStep + 1 ) );
-			tryAutoAdvance( false );
-			return;
-		}
-
-		submitForm( () => {
-			if ( getState().form.errors && ! isArray( getState().form.errors ) ) {
-				dispatch( NoticeActions.errorNotice( getState().form.errors, {
-					duration: 7000,
-				} ) );
-				return;
-			}
-
-			if ( getStepFormErrors( getState(), formSchema, formLayout ).length ) {
-				return;
-			}
-
-			if ( isManualAction && currentStepLayout.confirmation_flag ) {
-				dispatch( FormValueActions.updateField( currentStepLayout.confirmation_flag, true ) );
-			}
-
-			dispatch( goToStep( currentStep + 1 ) );
+	dispatch( {
+		type: GO_TO_STEP,
+		step: stepIndex,
+	} );
+};
 
-			tryAutoAdvance( false );
-		} );
-	};
+export const resetFlow = () => ( dispatch ) => {
+	dispatch( setFormProperty( 'pristine', true ) );
+	dispatch( goToStep( -1 ) );
+	dispatch( setFormProperty( 'acceptSuggestion', undefined ) );
```

Comment:
Instead of acceptSuggestion, if we support N addresses this should probably be something like 'selectDestinationAddress' and 0